bin/build_cmd.ml

```diff
@@ -215,3 +215,70 @@ let fmt =
     run_build_command ~common ~config ~request
   in
   Cmd.v (Cmd.info "fmt" ~doc ~man ~envs:Common.envs) term
+
+let doc =
+  let doc = "Build and view the documentation of an OCaml project" in
+  let man =
+    [ `S "DESCRIPTION"
+    ; `P
+        {|$(b,dune ocaml doc) builds and then opens the documention of an OCaml project in the users default browser.|}
+    ; `Blocks Common.help_secs
+    ]
+  in
+  let info = Cmd.info "doc" ~doc ~man in
+  let term =
+    let+ common = Common.term in
+    let config = Common.init common in
+    let request (setup : Import.Main.build_system) =
+      let dir = Path.(relative root) (Common.prefix_target common ".") in
+      let open Action_builder.O in
+      let+ () =
+        Alias.in_dir
+          ~name:(Dune_engine.Alias.Name.of_string "doc")
+          ~recursive:true ~contexts:setup.contexts dir
+        |> Alias.request
+      in
+      let is_default ctx =
+        ctx |> Context.name |> Dune_engine.Context_name.is_default
+      in
+      let doc_ctx = List.find_exn setup.contexts ~f:is_default in
+      let toplevel_index_path = Dune_rules.Odoc.Paths.toplevel_index doc_ctx in
+      let absolute_toplevel_index_path =
+        Path.(toplevel_index_path |> build |> to_absolute_filename)
+      in
+      Dune_console.print
+        [ Pp.textf "Docs built. Index can be found here: %s\n"
+            absolute_toplevel_index_path
+        ];
+
+      let url = Printf.sprintf {|file://%s|} absolute_toplevel_index_path in
+      let cmd =
+        let open Option.O in
+        let path = Env_path.path Env.initial in
+        let* cmd_name, args =
+          match Ocaml_config.system doc_ctx.ocaml_config with
```

Comment:
We now have a `Platform` module that you can use

Reply:
Oh that's very nice, thanks !